File: backend/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;


namespace prid_2324_a12.Models;

public class Question {
    [Key]
    public int Id {get;set;}
    [ForeignKey(nameof(QuizId))]  [JsonIgnore]
    public Quiz Quiz{get;set;}= null!;
    public int QuizId{get;set;}
    
    public int Order{get;set;}
    public string Body{get;set;}= null!;

    public virtual ICollection<Answer> Answers {get;set;} = new HashSet<Answer>();
    
    public virtual ICollection<Solution> Solutions {get;set;} = new HashSet<Solution>();


}

File: frontend/src/app/models/question.ts
import { Type } from "class-transformer";
import 'reflect-metadata';
import { Quiz } from "./quiz";
import { Solution } from "./solution";
import { Database } from "./database";


export class Question{
    id?: number;
    order?: number;
    body?: string;
    quizId?: number;
    quizName?: string;
    database?: Database;
    solutions?: Solution[];
    quizIsTest?: boolean;
    openedPanel?:boolean= false;
}